File: core/baker-annotations/src/main/java/com/ing/baker/recipe/annotations/RequiresIngredient.java
package com.ing.baker.recipe.annotations;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * An annotation to be added to an argument of an interaction to show what ingredient should be used here.
 */
@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.PARAMETER)
public @interface RequiresIngredient {
    String value() default "";
}

File: core/baker-annotations/src/main/kotlin/com/ing/baker/recipe/annotations/RequiresIngredient.kt
package com.ing.baker.recipe.annotations

import javax.inject.Qualifier

/**
 * An annotation to be added to an argument of an interaction to show what ingredient should be used here.
 */
@Qualifier
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class RequiresIngredient(
    val value: String = ""
)
